pkg/controller/common/container/defaulter.go

```diff
@@ -0,0 +1,175 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package container
+
+import (
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Defaulter ensures that values are set if none exists in the base container.
+type Defaulter interface {
```

Comment:
is there a use case for this interface vs. just using a public `Defaulter` struct directly with no interface defined?

Reply:
The interface helped me to drive the development, happy to remove it.

Reply:
I think I would remove it, but it's up to you, I really don't mind keeping it.